src/server/tests.rs

```diff
@@ -2054,17 +2054,41 @@ csv.from(
     let result =
         server.completion(params.clone()).await.unwrap().unwrap();
 
-    let items = match result {
-        lsp::CompletionResponse::List(l) => l.items,
-        _ => unreachable!(),
-    };
-
-    let labels: Vec<&str> =
-        items.iter().map(|item| item.label.as_str()).collect();
-
-    let expected = vec!["csv", "file", "mode", "url"];
-
-    assert_eq!(expected, labels);
+    expect![[r#"
+        {
+          "isIncomplete": false,
+          "items": [
+            {
+              "label": "csv",
+              "kind": 5,
+              "detail": "string",
+              "insertText": "csv: ",
+              "insertTextFormat": 2
+            },
+            {
+              "label": "file",
+              "kind": 5,
+              "detail": "string",
+              "insertText": "file: ",
+              "insertTextFormat": 2
+            },
+            {
+              "label": "mode",
+              "kind": 5,
+              "detail": "string",
+              "insertText": "mode: ",
+              "insertTextFormat": 2
+            },
+            {
+              "label": "url",
+              "kind": 5,
+              "detail": "string",
+              "insertText": "url: ",
+              "insertTextFormat": 2
+            }
+          ]
+        }"#]]
+    .assert_eq(&serde_json::to_string_pretty(&result).unwrap());
```

Comment:
I'm not familiar with this code, what are all these items representing?

Reply:
This uses https://docs.rs/expect-test/1.1.0/expect_test/ to assert that the result of the completion code results in the json literal here. By using expect-test it is much easier to update it if/when the output needs to change and I serialize to json instead of comparing against the `std::fmt::Debug` string since `lsp` structs contains a lot of optional fields which adds noise (the json representation skips those entirely).